SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/UserIdContext.cs

```diff
@@ -13,6 +13,9 @@ public class UserIdContext
     {
         private static readonly object UserIdLock = new object();
         private static UserIdContext _instanceField;
+        private string _userId;
+
+        private static readonly object UserIdContextLock = new object();
```

Comment:
```suggestion
        private readonly object UserIdContextLock = new object();
```
The static here is very confusing as we work at instance level.

Reply:
Should the 'static' here in the original code also be removed `private static readonly object UserIdLock = new object();` ?